### PR TITLE
Making context bag mandatory and exposed as readonly

### PR DESCRIPTION
**Pull request by danielmarbach**
This PR is part of https://github.com/Particular/FeatureDevelopment/issues/324 and addresses the following:

It aligns the `DispatchOptions` with the newly introduced options

@Particular/nservicebus please review.

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/nservicebus can someone review and merge if good?

File: src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Core/DefaultTimeoutManager.cs [hunk 1]
```diff
@@ -18,7 +19,7 @@ public void PushTimeout(TimeoutData timeout)
         {
             if (timeout.Time.AddSeconds(-1) <= DateTime.UtcNow)
             {
-                var sendOptions = new DispatchOptions(timeout.Destination,new AtomicWithReceiveOperation(), new List<DeliveryConstraint>());
+                var sendOptions = new DispatchOptions(timeout.Destination,new AtomicWithReceiveOperation(), new List<DeliveryConstraint>(), new ContextBag());
```
Comment (andreasohlund) [inline src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Core/DefaultTimeoutManager.cs]: We should flow the context from the incoming `TimeoutMessageProcessorBehavior` I'll create a separate pull for that

File: src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Hosting/Windows/TimeoutDispatcherProcessorBehavior.cs [hunk 2]
```diff
@@ -26,7 +27,7 @@ public override void Terminate(PhysicalMessageProcessingStageBehavior.Context co
 
             if (TimeoutsPersister.TryRemove(timeoutId, out timeoutData))
             {
-                var sendOptions = new DispatchOptions(timeoutData.Destination, new AtomicWithReceiveOperation(), new List<DeliveryConstraint>());
+                var sendOptions = new DispatchOptions(timeoutData.Destination,new AtomicWithReceiveOperation(), new List<DeliveryConstraint>(), new ContextBag());
```
Comment (andreasohlund) [inline src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Hosting/Windows/TimeoutDispatcherProcessorBehavior.cs]: Same here, I will include this in the new pull mentioned

File: src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Core/DefaultTimeoutManager.cs (hunk 1 above)
Comment (danielmarbach) [inline src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Core/DefaultTimeoutManager.cs]: :+1:

File: src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Hosting/Windows/TimeoutDispatcherProcessorBehavior.cs (hunk 2 above)
Comment (danielmarbach) [inline src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Hosting/Windows/TimeoutDispatcherProcessorBehavior.cs]: :+1:

Comment (andreasohlund) [inline src/NServiceBus.Core/DelayedDelivery/TimeoutManager/Hosting/Windows/TimeoutDispatcherProcessorBehavior.cs]: Still wip but started here https://github.com/Particular/NServiceBus/pull/2806

**Pull request merged**